### PR TITLE
Move initial user setup from hash data to env var plus token

### PR DESCRIPTION
**Pull request by john-metabase**
Using most of the code from the base64 approach, I tried using the env var `MB_USER_DEFAULTS` for supplying initial user details at deployment time. This avoids putting any user data in URLs. The `/setup#...` URL still uses the hash but the value should just be the configured token rather than the full encoded data.

The new `/api/setup/user_defaults` endpoint is authenticated by the `token` parameter which must match the configured value (note: this is completely separate from the runtime-generated setup token).

The structure of the env var JSON is exactly like the JSON in #18471 with the addition of a top-level `token` key, e.g.:

```
{
  user: {
    first_name: "",
    last_name: "",
    email: "",
    site_name: ""
  },
  token: "..."
}
```

### CODE REVIEW

**Review by noahmoss — COMMENTED** [review]
Backend parts lgtm with a couple non-blocking comments. I'll let @ranquild sign off on the frontend changes.

File: frontend/src/metabase/setup/components/Setup.jsx [hunk 1]
```diff
@@ -56,9 +58,9 @@ export default class Setup extends Component {
     trackStructEvent("Setup", "Welcome");
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     this.setDefaultLanguage();
-    this.setDefaultDetails();
+    await this.setDefaultDetails();
```
Comment (ranquild) [inline frontend/src/metabase/setup/components/Setup.jsx:63]: trackStepSeen relies on the current `props`. Could you move `await this.setDefaultDetails` below it?

File: frontend/src/metabase/setup/components/Setup.jsx [hunk 2]
```diff
@@ -81,14 +83,11 @@ export default class Setup extends Component {
...
   }
 
-  setDefaultDetails() {
-    const { hash } = this.props.location;
-
-    try {
-      const userDetails = hash && JSON.parse(b64hash_to_utf8(hash));
+  async setDefaultDetails() {
+    const token = this.props.location.hash.replace(/^#/, "");
+    if (0 < token.length) {
```
Comment (ranquild) [inline frontend/src/metabase/setup/components/Setup.jsx:88]: nit: `if (token) {`

**Review by ranquild — CHANGES_REQUESTED** [review]
Waiting for the `trackStepSeen` fix, apart from that lgtm

**Review by ranquild — APPROVED** [review]

**Comment by ranquild** [comment]
Epic https://github.com/metabase/metabase/pull/18471

**Pull request merged**